sql/src/main/java/org/apache/druid/sql/calcite/schema/MetadataSegmentView.java

```diff
@@ -201,32 +197,41 @@ private void poll()
       query = "/druid/coordinator/v1/metadata/segments?includeOvershadowedStatus&" + sb;
     }
     Request request;
+    InputStreamFullResponseHolder responseHolder;
     try {
       request = coordinatorClient.makeRequest(
           HttpMethod.GET,
-          StringUtils.format(query),
-          false
+          StringUtils.format(query)
       );
+
+      responseHolder = coordinatorClient.go(
+          request,
+          new InputStreamFullResponseHandler()
+      );
+      if (responseHolder.getStatus().getCode() != HttpServletResponse.SC_OK) {
+        throw new RE(
+            "Failed to talk to coordinator leader at [%s]. Error code[%d], description[%s].",
+            query,
+            responseHolder.getStatus().getCode(),
+            responseHolder.getStatus().getReasonPhrase()
+        );
+      }
     }
-    catch (IOException e) {
+    catch (IOException | InterruptedException e) {
       throw new RuntimeException(e);
     }
-    ListenableFuture<InputStream> future = coordinatorClient.goAsync(
-        request,
-        responseHandler
-    );
 
     final JavaType typeRef = jsonMapper.getTypeFactory().constructType(new TypeReference<SegmentWithOvershadowedStatus>()
     {
     });
     return new JsonParserIterator<>(
         typeRef,
-        future,
+        Futures.immediateFuture(responseHolder.getContent()),
         request.getUrl().toString(),
         null,
         request.getUrl().getHost(),
         jsonMapper,
-        responseHandler
+        null
```

Comment:
can we remove this param now

Reply:
I didn't know where else it was used, if it is not used anywhere then we can remove  it yes.

Reply:
removed